hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -642,10 +642,12 @@ case class TableLeftJoinRightDistinct(left: TableIR, right: TableIR, root: Strin
     val leftValue = left.execute(hc)
     val rightValue = right.execute(hc)
 
+    val joinKey = left.typ.key.length
+    assert(joinKey == right.typ.key.length)
```

Comment:
If we're going to require equality here, shouldn't we just require equality in the class in general?

Reply:
ah, this is wrong. pushing up a fix.